starknet_py/cairo/serialization/factory.py

```diff
@@ -0,0 +1,116 @@
+from __future__ import annotations
+from collections import OrderedDict
+from typing import Dict
+
+from starknet_py.cairo.serialization.data_serializers.cairo_data_serializer import (
+    CairoDataSerializer,
+)
+from starknet_py.cairo.serialization.data_serializers.felt_serializer import (
+    FeltSerializer,
+)
+from starknet_py.cairo.serialization.data_serializers.payload_serializer import (
+    PayloadSerializer,
+)
+from starknet_py.cairo.serialization.data_serializers.struct_serializer import (
+    StructSerializer,
+)
+from starknet_py.cairo.serialization.data_serializers.named_tuple_serializer import (
+    NamedTupleSerializer,
+)
+from starknet_py.cairo.serialization.data_serializers.tuple_serializer import (
+    TupleSerializer,
+)
+from starknet_py.cairo.serialization.data_serializers.uint256_serializer import (
+    Uint256Serializer,
+)
+from starknet_py.cairo.serialization.function_serialization_adapter import (
+    FunctionSerializationAdapter,
+)
+from starknet_py.net.models.abi.model import Abi
+from starknet_py.cairo.serialization.errors import InvalidTypeException
+
+from starknet_py.cairo.data_types import (
+    CairoType,
+    StructType,
+    FeltType,
+    ArrayType,
+    TupleType,
+    NamedTupleType,
+)
+
+from starknet_py.cairo.serialization.data_serializers.array_serializer import (
+    ArraySerializer,
+)
+
+_uint256_type = StructType("Uint256", OrderedDict(low=FeltType(), high=FeltType()))
+
+
+def serializer_for_type(cairo_type: CairoType) -> CairoDataSerializer:
+    """
+    Create a serializer for cairo type.
+
+    :param cairo_type: CairoType.
+    :return: CairoDataSerializer.
+    """
+    if isinstance(cairo_type, FeltType):
+        return FeltSerializer()
+
+    if isinstance(cairo_type, StructType):
+        # Special case: Uint256 is represented as struct
+        if cairo_type == _uint256_type:
+            return Uint256Serializer()
+
+        return StructSerializer(
+            OrderedDict(
+                (name, serializer_for_type(member_type))
+                for name, member_type in cairo_type.types.items()
+            )
+        )
+
+    if isinstance(cairo_type, ArrayType):
+        return ArraySerializer(serializer_for_type(cairo_type.inner_type))
+
+    if isinstance(cairo_type, TupleType):
+        return TupleSerializer(
+            [serializer_for_type(member) for member in cairo_type.types]
+        )
+
+    if isinstance(cairo_type, NamedTupleType):
+        return NamedTupleSerializer(
+            OrderedDict(
+                (name, serializer_for_type(member_type))
+                for name, member_type in cairo_type.types.items()
+            )
+        )
+
+    raise InvalidTypeException(f"Received unknown Cairo type '{cairo_type}'.")
+
+
+# We don't want to require users to use OrderedDict. Regular python requires order since python 3.7.
```

Comment:
I wonder if it isn't a bit error prone. Would it be possible for someone to use something different that CPython that doesn't enforce order and have this function fail? 

I'm not sure if order is enforced by the language standard or is just a feature of default implementation.

Reply:
https://stackoverflow.com/a/39980744

python requires order since python 3.7.

Reply:
Right, it's fine to use dict then.